gateway/src/apicast/policy/3scale_batcher/auths_cache.lua

```diff
@@ -0,0 +1,67 @@
+local keys_helper = require('apicast.policy.3scale_batcher.keys_helper')
+
+local re = require('ngx.re')
+local re_split = re.split
+
+local setmetatable = setmetatable
+local format = string.format
+local tonumber = tonumber
+
+local _M = {}
+
+local mt = { __index = _M }
+
+--- Initialize a cache for authorizations.
+-- @tparam storage ngx.shared.dict Shared dict to store the authorizations
+-- @tparam ttl integer TTL for the cached authorizations
+-- @treturn AuthsCache New cache for authorizations
+function _M.new(storage, ttl)
+  local self = setmetatable({}, mt)
+  self.storage = storage
+  self.ttl = ttl
+  return self
+end
+
+local function value_to_cache(auth_status, rejection_reason)
+  if rejection_reason then
+    return format("%s:%s", auth_status, rejection_reason)
+  else
+    return auth_status
+  end
+end
+
+--- Get a cached authorization.
+-- @tparam service_id string Service ID
+-- @tparam credentials table The keys of the table are the credential type
+--   (user_key, app_id or access_token) and the values are the credentials
+-- @tparam usage Usage The usage to authorize
+-- @treturn table The table has a "status" and a "rejection_reason"
+function _M:get(service_id, credentials, usage)
+  local key = keys_helper.key_for_cached_auth(service_id, credentials, usage)
+  local cached_value = self.storage:get(key)
+
+  if not cached_value then return nil end
+
+  local split_val = re_split(cached_value, ':', 'oj')
```

Comment:
Might be good to set `max` of the `re.split` to `2`
https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/re.md#split

Reply:
Done.